### ISSUE TITLE
Excluding containers exclude the entire pod from scan

### ISSUE THREAD

**Issue by sebastienthirion** [derailed/popeye#134]
<img src="https://raw.githubusercontent.com/derailed/popeye/master/assets/popeye_bug.png" align="right" width="100" height="auto"/>

<br/>
<br/>
<br/>

**Describe the bug**
I'm trying to exclude some containers from pods scan following the [sample configuration file](https://github.com/derailed/popeye#the-spinachyaml-configuration), specifically `istio-proxy` and `istio-init` containers because we use Istio as our service mesh, eg:
```
popeye:
  excludes:
    v1/pods:
      - name: rx:.*/.*
        containers:
          - istio-proxy
          - istio-init
```

The problem is that the entire pod is marked as ✅ even if issues should be reported for the pod itself and other containers inside the pod. When popeye is used configless, those issues are correctly reported.

**To Reproduce**
1. Have Istio Service Mesh operational inside the cluster
2. Have the following `nginx.yaml` file:
    ```yaml
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: nginx
    spec:
      selector:
        matchLabels:
          app: nginx
      replicas: 1
      template:
        metadata:
          labels:
            app: nginx
        spec:
          containers:
          - name: nginx
            image: nginx
    ```
3. Have the following `spinach.yml` file:
   ```yaml
   popeye:
    excludes:
      v1/pods:
        - name: rx:.*/.*
          containers:
            - istio-proxy
            - istio-init
   ```
4. Deploy the `Nginx` example with Istio sidecar:
    ```bash
    $ istioctl kube-inject -f nginx.yaml | kubectl apply -f -
    deployment.apps/nginx created
    ```
5. Use popeye configless. Issues are correctly reported:
   ```
   $ popeye -n default -s pod

    PODS (1 SCANNED)                                                 💥 1 😱 0 🔊 0 ✅ 0 0٪
    ┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅
    · default/nginx-994c9854c-n6mfj........................................................💥
        🔊 [POP-206] No PodDisruptionBudget defined.
        😱 [POP-300] Using "default" ServiceAccount.
        😱 [POP-301] Connects to API Server? ServiceAccount token is mounted.
        😱 [POP-302] Pod could be running as root user. Check SecurityContext/Image.
        🐳 istio-init
          😱 [POP-306] Container could be running as root user. Check SecurityContext/Image.
        🐳 istio-proxy
          😱 [POP-103] No liveness probe.
          🔊 [POP-105] Readiness probe uses a port#, prefer a named port.
        🐳 nginx
          💥 [POP-100] Untagged docker image in use.
          😱 [POP-106] No resources requests/limits defined.
          😱 [POP-102] No probes defined.
          😱 [POP-306] Container could be running as root user. Check SecurityContext/Image.
   ```
6. Use popeye with the `spinach.yml` example. Issues from the `nginx` container and the pod itself are missing:
   ```
   $ popeye -f spinach.yml -n default -s pod

   PODS (1 SCANNED)                                                💥 0 😱 0 🔊 0 ✅ 1 100٪
    ┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅
    · default/nginx-994c9854c-n6mfj........................................................✅
   ```

**Expected behavior**
Only the issues from the excluded containers should be excluded from the scan. Eg:
```
$ popeye -f spinach.yml -n default -s pod
PODS (1 SCANNED)                                                 💥 1 😱 0 🔊 0 ✅ 0 0٪
┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅
· default/nginx-994c9854c-n6mfj........................................................💥
    🔊 [POP-206] No PodDisruptionBudget defined.
    😱 [POP-300] Using "default" ServiceAccount.
    😱 [POP-301] Connects to API Server? ServiceAccount token is mounted.
    😱 [POP-302] Pod could be running as root user. Check SecurityContext/Image.
    🐳 nginx
        💥 [POP-100] Untagged docker image in use.
        😱 [POP-106] No resources requests/limits defined.
        😱 [POP-102] No probes defined.
        😱 [POP-306] Container could be running as root user. Check SecurityContext/Image.
```

**Versions (please complete the following information):**
 - OS: AKS
 - Popeye 0.8.10
 - K8s 1.18.8

**Comment by igoooor** [comment]
I confirm I'm facing the same issue.

**Comment by derailed** [comment]
@sebastienthirion @igoooor Thank you both for reporting this. I'll take a look and will fix in the next drop

**Comment by derailed** [comment]
@sebastienthirion @igoooor - Fixed v0.9.2!
